lib/worker/options.cjs

```diff
@@ -14,4 +20,11 @@ exports.set = newOptions => {
 	}
 
 	options = newOptions;
+	if (options.chalkOptions) {
+		if (setChalk) {
+			setChalk(options.chalkOptions);
+		} else {
+			chalk.then(chalk => chalk.set(options.chalkOptions));
+		}
```

Comment:
This is from 1b64bff382e00f85676aea888ab762b2ecc2d815, an independent experiment that can be reverted if deemed unsuccessful.
> This _seems_ to work, but the resulting interaction between CommonJS and ESM
feels a bit Zalgo-y. It may be better for modules to keep setting global
options and chalk options in parallel while the former remains CommonJS,
but I wanted to see what simplification would look like.

Reply:
Is this needed for this PR? Can we discuss it in a separate PR? It's all a little complicated I know, do we need to change it? What do you think the benefits are?

Reply:
I'd prefer to keep it, because otherwise the new tests don't correspond with reality (because worker output does not consistently respect color settings) and will undergo a fair amount of churn. But if you feel strongly that it should be separated, I can do that.

Reply:
I had a go at removing it, and other than `test-tap/assert.js` requiring an extra `stripAnsi()` there's no impact. I think I prefer the "set once" approach rather than having code react to changes.